presto-main/src/main/java/io/prestosql/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -558,9 +558,9 @@ protected Type visitSearchedCaseExpression(SearchedCaseExpression node, Stackabl
         @Override
         protected Type visitSimpleCaseExpression(SimpleCaseExpression node, StackableAstVisitorContext<Context> context)
         {
-            for (WhenClause whenClause : node.getWhenClauses()) {
-                coerceToSingleType(context, whenClause, "CASE operand type does not match WHEN clause operand type: %s vs %s", node.getOperand(), whenClause.getOperand());
-            }
+            coerceToSingleType(context,
```

Comment:
The existing code already checks that every operand of a `when` clause can be coerced to a common type with the operand of the `case` clause. Can you give an example of what issue this is intending to fix?

Reply:
operand is 8, first whenClause coerce operand 8 to double type, second whenCaluse coerce operand 8 to real type, overwrite previous coerce type.